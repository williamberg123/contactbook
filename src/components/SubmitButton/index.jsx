import PropTypes from 'prop-types';
import StyledSubmitButton from './styles';

export default function SubmitButton({ children }) {
	return (
		<StyledSubmitButton type="submit">{children}</StyledSubmitButton>
	);
}

SubmitButton.propTypes = {
	children: PropTypes.string.isRequired,
};
