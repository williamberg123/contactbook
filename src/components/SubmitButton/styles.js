import styled from 'styled-components';

const StyledSubmitButton = styled.button`
	background-color: #00bc5e;
	border: none;
	border-radius: 5px;
	color: white;
	font-weight: bold;
	padding: 15px;
	font-size: 1rem;
	text-align: center;
	cursor: pointer;

	&:hover {
		background-color: #4aef9d;
		transition: 0.3s;
	}
`;

export default StyledSubmitButton;
