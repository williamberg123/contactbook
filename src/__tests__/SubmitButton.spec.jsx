import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import SubmitButton from '../components/SubmitButton';

describe('<SubmitButton />', () => {
	it('should render SubmitButton with the text correctly', () => {
		render(<SubmitButton>Este é o botão de submit</SubmitButton>);
		expect(screen.getByRole('button', { name: /este é o botão de submit/i })).toBeInTheDocument();
		expect(screen.getByRole('button', { name: /este é o botão de submit/i })).toHaveAttribute('type', 'submit');
	});

	it('should render SubmitButton with the text correctly', () => {
		render(<SubmitButton>Este é o botão de submit</SubmitButton>);
		expect(screen.getByRole('button', { name: /este é o botão de submit/i })).toMatchSnapshot();
	});
});
