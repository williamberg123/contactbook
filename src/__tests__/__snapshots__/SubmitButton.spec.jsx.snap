// Vitest Snapshot v1

exports[`<SubmitButton /> > should render SubmitButton with the text correctly 1`] = `
.c0 {
  background-color: #00bc5e;
  border: none;
  border-radius: 5px;
  color: white;
  font-weight: bold;
  padding: 15px;
  font-size: 1rem;
  text-align: center;
  cursor: pointer;
}

.c0:hover {
  background-color: #4aef9d;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

<button
  class="c0"
  type="submit"
>
  Este é o botão de submit
</button>
`;
